src/c-writer.cc

```diff
@@ -1296,12 +1449,15 @@ void CWriter::Write(const Func& func) {
 
   WriteStackVarDeclarations();
 
-  std::unique_ptr<OutputBuffer> buf = func_stream_.ReleaseOutputBuffer();
-  stream_->WriteData(buf->data.data(), buf->data.size());
+  for (auto& [condition, stream] : func_sections_) {
+    std::unique_ptr<OutputBuffer> buf = stream.ReleaseOutputBuffer();
+    if (condition.empty() || func_includes_.count(condition)) {
+      stream_->WriteData(buf->data.data(), buf->data.size());
+    }
+  }
```

Comment:
Can you explain this part?

Reply:
Yes, it's part of https://github.com/WebAssembly/wabt/pull/1930#discussion_r919555274 (this is where we print out the sections of the function, and especially the sections that save caught exceptions to be rethrown later -- if there was a rethrow instruction that identifies that caught exception).